File: server/src/main/java/by/brawl/ws/service/GameService.java
package by.brawl.ws.service;

import by.brawl.entity.Squad;
import by.brawl.ws.holder.gamesession.GameSession;

import java.util.List;

public interface GameService {

    void createTwoPlayersGame(GameSession firstSession,
                              GameSession secondSession,
                              Squad firstSquad,
                              Squad secondSquad);

    void setHeroesPositions(GameSession session,
                            List<String> heroesIds);

    void castSpell(GameSession session,
                   String spellId,
                   Integer victimPosition,
                   Boolean forEnemy);
}

File: server/src/main/java/by/brawl/ws/service/GameService.kt
package by.brawl.ws.service

import by.brawl.entity.Squad
import by.brawl.ws.holder.gamesession.GameSession

interface GameService {

    fun createTwoPlayersGame(firstSession: GameSession,
                             secondSession: GameSession,
                             firstSquad: Squad,
                             secondSquad: Squad)

    fun setHeroesPositions(session: GameSession,
                           battleHeroesIds: List<String>)
}